### PR TITLE
[SPARK-44782][INFRA] Adjust PR template to Generative Tooling Guidance recommendations

### PR DESCRIPTION
**Pull request by zero323**
### What changes were proposed in this pull request?

This PR adds _Was this patch authored or co-authored using generative AI tooling?_ section to the PR template.

### Why are the changes needed?

To reflect recommendations of the [ASF Generative Tooling Guidance](https://www.apache.org/legal/generative-tooling.html).

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

Manual inspection.

### CODE REVIEW

**Comment by zhengruifeng** [comment]
I guess this should be documented in https://spark.apache.org/developer-tools.html instead of PR template?

cc @HyukjinKwon @gatorsmile @srowen

**Comment by srowen** [comment]
It should be in both places.

It's interesting. The typical CONTRIBUTING.md text that we have says, "When you contribute code, you affirm that the contribution is your original work", which seems at odds with the ASF guidance. I assume we are counting AI-assisted code as original work.

File: .github/PULL_REQUEST_TEMPLATE [hunk 1]
```diff
@@ -47,3 +47,12 @@ If it was tested in a way different from regular unit tests, please clarify how
 If tests were not added, please describe why they were not added and/or why it was difficult to add.
 If benchmark tests were added, please run the benchmarks in GitHub Actions for the consistent environment, and the instructions could accord to: https://spark.apache.org/developer-tools.html#github-workflow-benchmarks.
 -->
+
+
+### Was this patch authored or co-authored using generative AI tooling?
+<!--
+If a generative AI tooling has been used in the process of authoring this patch, please include
+frase 'Generated-by: ' followed by the name of the tool and its version.
```
Comment (yaooqinn) [inline .github/PULL_REQUEST_TEMPLATE:55]:
```suggestion
phrase 'Generated-by: ' followed by the name of the tool and its version.
```

File: .github/PULL_REQUEST_TEMPLATE [hunk 2]
```diff
@@ -47,3 +47,12 @@ If it was tested in a way different from regular unit tests, please clarify how
 If tests were not added, please describe why they were not added and/or why it was difficult to add.
 If benchmark tests were added, please run the benchmarks in GitHub Actions for the consistent environment, and the instructions could accord to: https://spark.apache.org/developer-tools.html#github-workflow-benchmarks.
 -->
+
+
+### Was this patch authored or co-authored using generative AI tooling?
+<!--
+If a generative AI tooling has been used in the process of authoring this patch, please include
```
Comment (srowen) [inline .github/PULL_REQUEST_TEMPLATE:54]: Change to "If generative AI ..."
Change to "please include the"

**Review by srowen — CHANGES_REQUESTED** [review]

**Comment by srowen** [comment]
Merged to master (the failure can't be related)

**Pull request closed without merging**